### DISCUSSION TITLE
Problem with Filters - always empty

### DISCUSSION THREAD

**Discussion by lfontana1984** [microsoft-search/pnp-modern-search#2067] [Q&A]
Dears,
we cannot use the "filters" web part because it is not working at all.
We add the search result in the page and everything is working fine.
When we add the filters, we always get "No available filter to display".

Is it possible to understand what the problem is?

Thanks,

Luig
![2022-05-05_15h06_33](https://user-images.githubusercontent.com/104991695/166929156-ba489ae5-1e3e-4c73-918e-0e1e67e2e2c2.png)
i

**Comment by FranckyC** [comment]
Hi @lfontana1984, with no details about the configuration you've made, how do you expect us to solve your issue? Please follow the issue template and give us all required information so we can help you. With no details we'll close your issue.

Also did you have a look at the documentation to configure the Web Part? https://microsoft-search.github.io/pnp-modern-search/usage/search-filters/

**Comment by lfontana1984** [comment]
Hi,
I added no details because I don't have big details to add unfortunately.
We install the last version of the PNP search.
We create a simple test team site and add 2 web parts:
1) Search result: without any kind of customization, so only a "add" and save
2) Search Filter: only a simple filter connected to the search result and a field configured. No matter which field we use, we never have a result.

Here attached some screenshots.
Let me know if you need more info
![2022-05-05_15h28_43](https://user-images.githubusercontent.com/104991695/166936349-bdeb5c8b-c6ca-44be-bbee-ad5f3d916e5d.png)
![2022-05-05_15h29_34](https://user-images.githubusercontent.com/104991695/166936375-4e67e7b3-b22d-4583-beda-6e673a47314e.png)
![2022-05-05_15h28_20](https://user-images.githubusercontent.com/104991695/166936463-3af4a856-6d99-4ac2-8716-b04ba5608c02.png)
.

**Reply by FranckyC** [reply]
Connection is two ways, You need to connect search filters WP from the search results WP as well.

**Reply by lfontana1984** [reply]
OK, now the standard is working as expected.
If I add this part to the query in the result WP:
"CMAP-CompanyName":{inputQueryText}
NOthing will work again.

Any advice on this?

![2022-05-05_17h42_23](https://user-images.githubusercontent.com/104991695/166960950-d0c80f4d-eabc-4f1f-a395-bb1ae48220dc.png)

**Reply by FranckyC** [reply]
Your query seems incorrect. Try without quotes.

**Reply by FranckyC** [reply]
Also `{inputQueryText}` uses the search box value OR the defaut value in the search results WP. If no value, no results.

**Reply by wobba** [reply]
I do not recommend using `-` in custom managed property names or aliases. Stick with a-z/A-Z/0-9.

**Reply by lfontana1984** [reply]
> Your query seems incorrect. Try without quotes.

Hi, WIthout quotes it does not work (maybe due to the "-" character in the name of the property). The search result works properly with the above query if something is available in the query text (configured as a URL parameter). But the filter is totally not displayed. That's my issue.

> Also `{inputQueryText}` uses the search box value OR the defaut value in the search results WP. If no value, no results.

What does it mean?


> I do not recommend using `-` in custom managed property names or aliases. Stick with a-z/A-Z/0-9.

You're right. We will keep this in mind next time....

**Reply by lfontana1984** [reply]
Dears, any comment? @FranckyC

**Reply by lfontana1984** [reply]
We found that the problem was caused by the "-" in the property name. Remapping everything without the "-" everything starts working properly.

**Reply by wobba** [reply]
You can add an alias to the managed property without a hyphen as well to avoid re-mapping.
